drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1097 @@
+	}
+#endif
+
+	regdata |= FIELD_PREP(TMAG5273_CRC_EN_MSK, val);
+
+	switch (drv_cfg->temperature_coefficient) {
+	case TMAG5273_DT_TEMP_COEFF_NDBFE:
+		val = TMAG5273_MAGNET_TEMP_COEFF_NDBFE;
+		break;
+	case TMAG5273_DT_TEMP_COEFF_CERAMIC:
+		val = TMAG5273_MAGNET_TEMP_COEFF_CERAMIC;
+		break;
+	case TMAG5273_DT_TEMP_COEFF_NONE:
+		__fallthrough;
+	default:
+		val = TMAG5273_MAGNET_TEMP_COEFF_NONE;
+		break;
+	}
+
+	regdata |= FIELD_PREP(TMAG5273_MAG_TEMPCO_MSK, val);
+
+	switch (drv_cfg->averaging) {
+	case TMAG5273_DT_AVERAGING_2X:
+		val = TMAG5273_CONV_AVG_2;
+		break;
+	case TMAG5273_DT_AVERAGING_4X:
+		val = TMAG5273_CONV_AVG_4;
+		break;
+	case TMAG5273_DT_AVERAGING_8X:
+		val = TMAG5273_CONV_AVG_8;
+		break;
+	case TMAG5273_DT_AVERAGING_16X:
+		val = TMAG5273_CONV_AVG_16;
+		break;
+	case TMAG5273_DT_AVERAGING_32X:
+		val = TMAG5273_CONV_AVG_32;
+		break;
+	case TMAG5273_DT_AVERAGING_NONE:
+		__fallthrough;
+	default:
+		val = TMAG5273_CONV_AVG_1;
+		break;
+	}
+
+	drv_data->conversion_time_us = TMAG5273_T_CONVERSION_US(
+		(val), (drv_cfg->x_axis + drv_cfg->y_axis + drv_cfg->z_axis));
+
+	regdata |= FIELD_PREP(TMAG5273_CONV_AVG_MSK, val);
+	regdata |= FIELD_PREP(TMAG5273_I2C_READ_MSK, I2C_READ_MODE_STANDARD);
+
+	retval = i2c_reg_write_byte_dt(&drv_cfg->i2c, TMAG5273_REG_DEVICE_CONFIG_1, regdata);
+	if (retval < 0) {
+		LOG_ERR("error setting DEVICE_CONFIG_1");
+		return -EIO;
+	}
+
+	/* REG_DEVICE_CONFIG_2 */
+	regdata = 0;
+
+	val = (drv_cfg->low_noise_mode) ? TMAG5273_LOWNOISE_MODE : TMAG5273_LOWPOWER_MODE;
+	regdata |= FIELD_PREP(TMAG5273_LP_LN_MSK, val);
+
+	regdata |= FIELD_PREP(TMAG5273_I2C_GLITCH_FILTER_MSK, TMAG5273_I2C_GLITCH_FILTER_ON);
+
+	val = (drv_cfg->trigger_conv_via_int) ? TMAG5273_TRIGGER_MODE_INT
+					      : TMAG5273_TRIGGER_MODE_I2C;
+	regdata |= FIELD_PREP(TMAG5273_TRIGGER_MODE_MSK, val);
+
+	val = (drv_cfg->operation_mode == TMAG5273_DT_OPER_MODE_CONTINUOUS)
+		      ? TMAG5273_OPERATING_MODE_CONTINUOUS
+		      : TMAG5273_OPERATING_MODE_STANDBY;
+	regdata |= FIELD_PREP(TMAG5273_OPERATING_MODE_MSK, val);
+
+	retval = i2c_reg_write_byte_dt(&drv_cfg->i2c, TMAG5273_REG_DEVICE_CONFIG_2, regdata);
+	if (retval < 0) {
+		LOG_ERR("error setting DEVICE_CONFIG_2");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief sets the \c SENSOR_CONFIG_1 and \c SENSOR_CONFIG_2 registers
+ *
+ * @param drv_cfg configuration of the TMAG5273 instance
+ *
+ * @retval 0 if everything was okay
+ * @retval -EIO on communication errors
+ */
+static inline int tmag5273_init_sensor_settings(const struct tmag5273_config *drv_cfg)
+{
+	int retval;
+	uint8_t regdata;
+
+	/* REG_SENSOR_CONFIG_1 */
+	regdata = 0;
+	uint8_t val = ((drv_cfg->x_axis) ? TMAG5273_MAGNETIC_CH_X_EN : TMAG5273_MAGNETIC_CH_NONE) |
+		      ((drv_cfg->y_axis) ? TMAG5273_MAGNETIC_CH_Y_EN : TMAG5273_MAGNETIC_CH_NONE) |
+		      ((drv_cfg->z_axis) ? TMAG5273_MAGNETIC_CH_Z_EN : TMAG5273_MAGNETIC_CH_NONE);
+
+	regdata |= FIELD_PREP(TMAG5273_MAG_CH_EN_MSK, val);
```

Comment:
How about avoiding FIELD_PREP for these?

```c
if (dev_cfg->x_axis)) {
        regdata |= BIT(TMAG5273_MAG_CH_X_EN);
}
...
```

and so on. Would make the pattern way obvious, allow to drop few mask defines...